### PR TITLE
NACA0012: RST scripts, csv tables and new figures

### PR DESCRIPTION
**Pull request by XiaoyuYang1981**
@maciej-flexcompute These are updates for the CaseStudy on NACA0012 low speed airfoil. RST scripts with csv data table and new figures are updated. External links to the grid file, the mesh json and the solver json files, still need to be updated.

### CODE REVIEW

File: docs/source/examples/naca0012_caseStudy.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   
+.. |agr|  unicode:: U+003B1 .. GREEK SMALL LETTER ALPHA
+
+NACA 0012 Low Speed Airfoil
+===========================
+
+Introduction
+------------
+
+The purpose of this case study is to verify and validate the functionality of the Flow360 solver and the implementation of the Spalart-Allmaras (S-A) turbulence model, with respect to the NACA 0012 low speed airfoil validation case. This is one of the cases originally provided by `the NASA Langley Turbulence Modeling Resource database <https://turbmodels.larc.nasa.gov/naca0012_val.html>`_ with accurate and up-to-date information on widely-used turbulence models, as also described by `the ONERA-elsA website <http://elsa.onera.fr/TMR-0001/GENERATED.html#academic-naca0012-lowspeed>`_. Recent high-accuracy numerical investigations on asymptotic grid convergence behaivors are described on this `NASA-LaRC-TMR webpage <https://turbmodels.larc.nasa.gov/naca0012numerics_val.html>`_. Similar validation results of OVERFLOW were documented by `Jespersen et al <https://turbmodels.larc.nasa.gov/Papers/NAS_Technical_Report_NAS-2016-01.pdf>`_.
```
[the remark this answers is not in the document]
Comment (jbungenerflex) [reply docs/source/examples/naca0012_caseStudy.rst:14]: NAiver -> Navier TYPO

File: docs/source/examples/naca0012_caseStudy.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
...ven in the above NASA webpage, that of Family II at Level-4 (897 x 2 x 257) was used in this validation. The geometry and the mesh of this quasi-3D CFD model are shown in :numref:`Fig1_NACA0012GeoMsh`. As seen, the grid has a C-type topology, wrapping around the airfoil from the downstream farfield, around the lower surface to the upper, then back to the downstream farfield again. Also, in the wake region, the grid connects to itself in a point-matched manner. The chord length is scaled into unity, and the grid unit is taken to be meter. The origin of the grid coordinates is at the leading edge of the airfoil. The x-axis is in alignment with the chord line. Also, the grid is stretched for one spanwise cell in the negative y direction with the grid spacing of 1 meter. The original unstructured CGNS grid files downloaded from the NASA webpage are in the ADF data format and have to be converted into the HDF5, to meet :ref:`the grid requirements<manualMeshingFlow360Section>` of Flow360. For this validation case, the converted grid file with the HDF5 data structure is available `here <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/n0012familyII.4.hex.hdf5.cgns.tar.gz>`_.
```
Comment (jbungenerflex) [inline docs/source/examples/naca0012_caseStudy.rst:21]: I would reword htat phrase to be something like.
"From the three series...... available from the above NASA webpage (add link to it again), we used the .... grid for this validation."

Comment (jbungenerflex) [reply docs/source/examples/naca0012_caseStudy.rst:21]: Can you please add a quick line as to why you chose this particular mesh?

File: docs/source/examples/naca0012_caseStudy.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
...n dashed lines for the S-A model without and with the R/C correction, respectively. The corresponding CFL3D results are shown as the blue dashed and dash-dot lines. Experimental data are given as symbols. Notably, these measurements of :math:`C_p` were made at a lower Reynolds number :math:`Re_c = 2.88 \times 10^6` by `Gregory and O'Reilly <https://reports.aerade.cranfield.ac.uk/bitstream/handle/1826.2/3003/arc-rm-3726.pdf>`_. As seen from these figures, at all the three distinct lift conditions, the current predictions accurately capture the experimental data, and closely match the counterparts of the reference numerical results. Notably, as shown in :numref:`Fig3c_CpCf_AOA15` associated with :math:`\alpha = 15^o`, the :math:`C_f` value of the current result with the S-A model approaches zero around :math:`x/c = 0.9078` on the upper surface of the airfoil near the trailing edge. This indicates separated mean flow occurs as observed from the above :numref:`Fig2c_Ma_mutRatio_AOA15`. For the current result with the SA-RC model, the separation point emerges around :math:`x/c = 0.8992`. These locations are at the upstream bounds of the suggested intervals as given on the NASA webpages.
```
Comment (jbungenerflex) [inline docs/source/examples/naca0012_caseStudy.rst:79]: replace "current CFD results " with "Flow360 results"

File: docs/source/examples/naca0012_caseStudy.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
...e CFD data. As seen, for all the tested operating conditions, the accuracy of current results are competitive. It is noted that, for most of these representative aerodynamic quantities, Flow360 provides bound values. It is suspected that this is due to the differences in the discretization, such as node-centered or cell-centered schemes, as well as convergence criteria, at least as seen from the :math:`C_l` values at :math:`\alpha = 0^o`. Besides, the reference data such as those CFL3D results were computed with the farfield point vortex correction as given by `Thomas and Salas <https://doi.org/10.2514/3.9394>`_, although an earlier typo issue was involved in the scaled airfoil formula. A different computational grid is used for the current cases. The corresponding data are also displayed in :numref:`Fig4_ClCd`, together with the experimental data measured by `Ladson <https://ntrs.nasa.gov/citations/19880019495>`_. Notably, in this figure, at low-to-moderate angles of attack, linear variation of :math:`C_l` with the best fit lift slope provided by `McCroskey <https://ntrs.nasa.gov/citations/19880002254>`_ is also shown. As seen, the current results agree well with these references.
```
Comment (jbungenerflex) [inline docs/source/examples/naca0012_caseStudy.rst:99]: please add 1 bar graphs for the alpha 0 Cd results and another 2 bar grapha for the alpha 10 Cl and Cd values. 
AS in 1 bar per CFD code. 
Also, "current results are competitive" => "Flow360 results are comparable"

File: docs/source/examples/naca0012_caseStudy.rst [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+.. _Fig4_ClCd:
+.. figure:: figures_NACA0012/NACA0012_Cl_Cd_Flow360_SA_upd1.png
+   :align: center
+
+   Comparisons of :math:`C_l` and :math:`C_d` at :math:`\alpha = 0^o`, :math:`10^o` and :math:`15^o`.
+   
+Remarks
+-------
+
+The farfield point vortex correction based on inviscid characteristic methods, such as the above, is involved in the above reference data, such as those typical CFL3D validation results. At the current accuracy level for validation with farfield boundaries about 500 chords away, its effects are inappreciable. But, the influence would emerge either at detailed accuracy levels or with a smaller computational domain, especially at high lift conditions, for example, as shown on this NASA webpage about `the effect of farfield boundary <https://turbmodels.larc.nasa.gov/naca0012_val_ffeffect.html>`_, as well as through numerical analysis of grid convergence.
```
Comment (jbungenerflex) [inline docs/source/examples/naca0012_caseStudy.rst:118]: These two paragraphs in the remarks are confusing. I don't understand what you are trying to say.

I think that the last paragraph should be reworded and used at the very beginning when you describe why you chose that mesh you chose.

**Review by jbungenerflex — COMMENTED** [review]
Good job, very impressive results.
Thank you

**Comment by XiaoyuYang1981** [comment]
> Good job, very impressive results. Thank you

First of all, thank you very much for your comments and efforts, let me look into these

File: docs/source/examples/naca0012_caseStudy.rst (hunk 4 above)
Comment (XiaoyuYang1981) [reply docs/source/examples/naca0012_caseStudy.rst:99]: > please add 1 bar graphs for the alpha 0 Cd results and another 2 bar grapha for the alpha 10 Cl and Cd values. AS in 1 bar per CFD code. Also, "current results are competitive" => "Flow360 results are comparable"

Hi Jim, not quite sure, but I am not fully understand the first part of your suggestions here. Please have a look at the updated script and let me know whether the issue was settled or not.

File: docs/source/examples/naca0012_caseStudy.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+.. _Fig3a_CpCf_AOA00:
```
[the remark this answers is not in the document]
Comment (XiaoyuYang1981) [reply docs/source/examples/naca0012_caseStudy.rst:81]: > Where is CFL3D data for Cp plot, it is in the legend but not visible in the graph. If it is hidden by Flow360 values, then I would either change size/type of CFL3D markers or put a note in figure legend saying the CFL3D Cp is hidden underneath Flow360.

The line width for CFL3D results was increased.

File: docs/source/examples/naca0012_caseStudy.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+Steady-state solutions were computed iteratively, utilizing 2nd-order schemes for both the RANS equations and the turbulence model equation. For the former, the CFL number was gradually ramped up to 100. Detailed Flow360 solver and mesh configuration files, `<Flow360.json> <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/NACA0012_FMLY2GL4_AOA15_SARC_2ndOrd_CFL100_Flow360.json>`_ and `<Flow360Mesh.json> <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/NACA0012_FMLY2_Flow360Mesh.json>`_, are given here as an example for the case with the S-A model and its Rotation/Curvature (R/C) correction at :math:`\alpha = 15^o`.
+
+Besides, according to :ref:`Flow360's definitions<Flow360json>` of the angle of attack :math:`\alpha`, the sideslip angle :math:`\beta` and the grid coordinates, the following correlations are imposed at the “Freestream” farfield boundaries.
```
[the remark this answers is not in the document]
Comment (XiaoyuYang1981) [reply docs/source/examples/naca0012_caseStudy.rst:45]: > I am not sure I understand the point of this paragraph. I would remove everything from "Besides .....values,etc." This is what a CFD code does. We don't need to discuss it. Does that make sense or am I missing something.

Indeed, understood. The reason for this was that when I started this case, I did not find any info about the orientations of alpha and beta w.r.t. the coordinates. The solver configuration part of this documentation did not explicitly state the directional arrangements of these angles. When starts new, one might not familiar with the conventions on this, and this might cause potential issue to setup a new mesh. I thought it might be helpful to keep this part. Let me know what do you say please.

[the remark this answers is not in the document]
Comment (maciej-flexcompute) [reply docs/source/examples/naca0012_caseStudy.rst:45]: @XiaoyuYang1981 can you please add formulations to alpha and beta docs and put link here?

[the remark this answers is not in the document]
Comment (maciej-flexcompute) [reply docs/source/examples/naca0012_caseStudy.rst:45]: @XiaoyuYang1981 On webUI when you use json generator you can check tooltips for alpha and beta, you can reuse them for alpha and beta docs

File: docs/source/examples.rst [hunk 8]
```diff
@@ -10,4 +10,4 @@ Case Studies
    examples/dpw4
    examples/om6_caseStudy
    examples/betDiskSteady_caseStudy
-
+   examples/naca0012_caseStudy
```
Comment (maciej-flexcompute) [inline docs/source/examples.rst:13]: lets put this case study as first

File: docs/source/examples/naca0012_caseStudy.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
+.. _naca0012_caseStudy:
+
+.. |deg|    unicode:: U+000B0 .. DEGREE SIGN
+   :ltrim:
+   
+.. |agr|  unicode:: U+003B1 .. GREEK SMALL LETTER ALPHA
+
+NACA 0012 Low Speed Airfoil
+===========================
+
```
Comment (maciej-flexcompute) [inline docs/source/examples/naca0012_caseStudy.rst:10]: add somewhere which version of code was used and add link to release-notes to this version

Comment (XiaoyuYang1981) [reply docs/source/examples/naca0012_caseStudy.rst:10]: the solver version was added to the end of the Remarks

File: docs/source/solverConfiguration/solverConfiguration.rst [hunk 10]
```diff
@@ -75,6 +75,17 @@ freestream
...
+According to Flow360's definitions of the angle of attack :math:`\alpha` and the sideslip angle :math:`\beta`, with respect to the grid coordinates, the following values of velocity components are imposed at a “Freestream” farfield boundary as defined in the next subsection.
+
+.. math::
+      :label: FreestreamBC
+	  
+	  U^*_{\infty} &= M_{\infty} \cdot cos(\beta) \cdot cos(\alpha) \\
+      V^*_{\infty} &= - M_{\infty} \cdot sin(\beta) \\
+      W^*_{\infty} &= M_{\infty} \cdot cos(\beta) \cdot sin(\alpha) 
+	  
+Where, the velocity components are non-dimensionalized by the freestream speed of sound :math:`C_{\infty}`. Also, the effects of these two angles are intrinsically taken into account by the solver in the computed :math:`C_l` and :math:`C_d` values, etc.
```
Comment (maciej-flexcompute) [inline docs/source/solverConfiguration/solverConfiguration.rst:88]: usually capital L, capital D are used in C_L, C_D.

File: docs/source/solverConfiguration/solverConfiguration.rst [hunk 11]
```diff
@@ -75,6 +75,17 @@ freestream
    :widths: 20, 20, 70
    :header-rows: 1
    :delim: @
+   
+According to Flow360's definitions of the angle of attack :math:`\alpha` and the sideslip angle :math:`\beta`, with respect to the grid coordinates, the following values of velocity components are imposed at a “Freestream” farfield boundary as defined in the next subsection.
+
+.. math::
```
Comment (maciej-flexcompute) [inline docs/source/solverConfiguration/solverConfiguration.rst:81]: add link to this equation from table, where alpha and beta are described

File: docs/source/examples/naca0012_caseStudy.rst [hunk 12]
```diff
@@ -32,26 +32,15 @@ As suggested by NASA-LaRC-TMR and ONERA-elsA, for this essentially incompressibl
...
-	  
-Where, the components of the velocity are non-dimensionalized by the freestream speed of sound :math:`C_{\infty}`. Also, the effects of these two angles are intrinsically taken into account by the solver in the computed :math:`C_l` and :math:`C_d` values, etc.
+Steady-state solutions were computed iteratively, utilizing 2nd-order schemes for both the RANS equations and the turbulence model equation. For the former, the CFL number was gradually ramped up to 100. Detailed Flow360 solver and mesh configuration files, `<Flow360.json> <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/NACA0012_FMLY2GL4_AOA15_SARC_2ndOrd_CFL100_Flow360.json>`_ and `<Flow360Mesh.json> <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/NACA0012_FMLY2_Flow360Mesh.json>`_, are given here as an example for the case with the S-A model and its Rotation/Curvature (R/C) correction at :math:`\alpha = 15^o`. Besides, the definitions of the angle of attack :math:`\alpha` and the sideslip angle :math:`\beta`, with respect to the grid coordinates, are given in :ref:`the solver configuration<Flow360json>` section of this documentation.
```
Comment (maciej-flexcompute) [inline docs/source/examples/naca0012_caseStudy.rst:43]: remove <> from Flow360.json and Flow360Mesh.json display names

File: docs/source/examples/naca0012_caseStudy.rst [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+.. figure:: figures_NACA0012/n0012familyII4hex_global.png
+   :align: center
+.. _Fig1_NACA0012GeoMsh:
+.. figure:: figures_NACA0012/n0012familyII4hex_local.png
+   :align: center
+
+   Quasi-3D CFD model of NACA 0012 airfoil (Family II, Grid Level-4).
+   
+As suggested by NASA-LaRC-TMR and ONERA-elsA, for this essentially incompressible flow, the freestream Mach number is :math:`M_{\infty} = 0.15`, the Reynolds number per chord length is :math:`Re_{c} = 6 \times 10^6`, and the reference temperature is :math:`T_{ref} = 300 K`. The turbulent flows past the airfoil were simulated at three angles of attack, i.e. :math:`\alpha = 0^o`, :math:`10^o` and :math:`15^o`. :numref:`Tab1_FlowConfig` summarizes the key flow configuration parameters for Flow360. General guidances on :ref:`the non-dimensionalization conventions<nondimensionalization_Flow360>` and :ref:`the model configurations<Flow360json>` are given in those sections of this online documentation.
```
Comment (maciej-flexcompute) [inline docs/source/examples/naca0012_caseStudy.rst:31]: this numref doesn't work. Where is the table?

File: docs/source/examples/naca0012_caseStudy.rst [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+
+Steady-state solutions were computed iteratively, utilizing 2nd-order schemes for both the RANS equations and the turbulence model equation. For the former, the CFL number was gradually ramped up to 100. Detailed Flow360 solver and mesh configuration files, `<Flow360.json> <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/NACA0012_FMLY2GL4_AOA15_SARC_2ndOrd_CFL100_Flow360.json>`_ and `<Flow360Mesh.json> <https://simcloud-public-1.s3.amazonaws.com/validation/naca0012/NACA0012_FMLY2_Flow360Mesh.json>`_, are given here as an example for the case with the S-A model and its Rotation/Curvature (R/C) correction at :math:`\alpha = 15^o`. Besides, the definitions of the angle of attack :math:`\alpha` and the sideslip angle :math:`\beta`, with respect to the grid coordinates, are given in :ref:`the solver configuration<Flow360json>` section of this documentation.
```
Comment (maciej-flexcompute) [inline docs/source/examples/naca0012_caseStudy.rst:43]: add link directly to the equation for alpha and beta

File: docs/source/examples/naca0012_caseStudy.rst (hunk 13 above)
Comment (XiaoyuYang1981) [reply docs/source/examples/naca0012_caseStudy.rst:31]: To main the same style of the file names, the name of the csv file for this table was updated as 'naca0012_tab1_keyFlowConfig.csv', but it seems that the RST script is not case sensitive and this info did not transfer through the Git either. So this has been rolled back to the earlier 'naca0012_tab1_keyflowconfig.csv'. Let me know if it is still not working.

File: docs/source/solverConfiguration/solverConfiguration.rst (hunk 11 above)
Comment (XiaoyuYang1981) [reply docs/source/solverConfiguration/solverConfiguration.rst:81]: Since in the section of the docs, tables were not given a label serial number, a short verbal indication is given instead.

**Review by maciej-flexcompute — APPROVED** [review]
